docs/book/component-guide/step-operators/kubernetes.md

```diff
@@ -0,0 +1,232 @@
+---
+description: Executing individual steps in Kubernetes Pods.
+---
+
+# Kubernetes Step Operator
+
+ZenML's Kubernetes step operator allows you to submit individual steps to be run on Kubernetes pods.
+
+### When to use it
+
+You should use the Kubernetes step operator if:
+
+* one or more steps of your pipeline require computing resources (CPU, GPU, memory) that are not provided by your orchestrator.
+* you have access to a Kubernetes cluster.
+
+### How to deploy it
+
+The Kubernetes step operator requires a Kubernetes cluster in order to run. There are many ways to deploy a Kubernetes cluster using different cloud providers or on your custom infrastructure, and we can't possibly cover all of them, but you can check out our cloud guide.
+
+### How to use it
+
+To use the Kubernetes step operator, we need:
+*   The ZenML `kubernetes` integration installed. If you haven't done so, run
+
+    ```shell
+    zenml integration install kubernetes
+    ```
+* A Kubernetes cluster [deployed](kubernetes.md#how-to-deploy-it)
+* Either [Docker](https://www.docker.com) installed and running or a remote [image builder](../image-builders/image-builders.md) in your stack.
+* A [remote artifact store](../artifact-stores/artifact-stores.md) as part of your stack. This is needed so that both your orchestration environment and Kubernetes Pods can read and write step artifacts. Check out the documentation page of the artifact store you want to use for more information on how to set that up and configure authentication for it.
+
+
+{% hint style="info" %}
+It is recommended that you set up [a Service Connector](../../how-to/auth-management/service-connectors-guide.md) and use it to connect the Kubernetes step operator to the Kubernetes cluster, especially if you are using a Kubernetes cluster managed by a cloud provider like AWS, GCP or Azure.
+{% endhint %}
+
+We can then register the step operator and use it in our stacks. This can be done in two ways:
+
+1.  If you have [a Service Connector](../../how-to/auth-management/service-connectors-guide.md) configured to access the remote Kubernetes cluster, you can [connect the stack component to the Service Connector](../../how-to/auth-management/service-connectors-guide.md#connect-stack-components-to-resources):
```

Comment:
link to k8s connector?

Reply:
I think in general the suggested way is to use a cloud service connector, not a kubernetes one.

I don't want to link to all of them, the guide should somehow cover that right?

Reply:
@schustmi still its logical to link it to k8s (for both comments) IMO

Reply:
It links to stuff that we don't want people to do. Why? I've never even registered one of these, and it probably won't even work easily for clusters in cloud providers, and requires a different set of config options.

Reply:
then id say lets link to all the clouds and make sure we're linking to really the right thing because otherwise it links to this ginormous guide